components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/RefreshGrantHandler.java

```diff
@@ -189,6 +192,23 @@ public OAuth2AccessTokenRespDTO issue(OAuthTokenReqMessageContext tokReqMsgCtx)
         return buildTokenResponse(tokReqMsgCtx, accessTokenBean);
     }
 
+    private void setSelectedAcrAndAuthTime(OAuthTokenReqMessageContext tokReqMsgCtx) {
+
+        RefreshTokenValidationDataDO oldAccessToken =
```

Comment:
We don't need these values for all the access tokens. Can you add these conditionally only when requested.